build/AzurePipelinesTemplates/WindowsAppSDK-RunTests-Steps.yml

```diff
@@ -0,0 +1,143 @@
+              certutil -addstore TrustedPeople $($cerFile.FullName)
+              certutil -addstore root $($cerFile.FullName)
+            }
+
+  - task: powerShell@2
+    displayName: 'Run dotnet installer'
+    inputs:
+      targetType: 'inline'
+      script: |
+            $(Build.SourcesDirectory)\packages\dotnet-windowsdesktop-runtime-installer.exe /quiet /install /norestart
+
+  - task: powerShell@2
+    displayName: 'Install VCLibs.Desktop'
+    inputs:
+      targetType: 'inline'
+      script: |
+            $package = "$(Build.SourcesDirectory)\packages\Microsoft.VCLibs.${{ parameters.buildPlatform }}.14.00.Desktop.appx"
+            Add-AppxPackage $package -ErrorAction SilentlyContinue
+
+  - task: powerShell@2
+    displayName: 'Install vc_redist'
+    inputs:
+      targetType: 'inline'
+      script: |
+            $(Build.SourcesDirectory)\packages\vc_redist.${{ parameters.buildPlatform }}.exe /quiet /install /norestart
+
+  - task: powershell@2
+    displayName: 'Run TE.Service'
+    inputs:
+      targetType: filePath
+      filePath: DevCheck.ps1
+      arguments: -NoInteractive -Offline -Verbose -CheckTAEFService
+      workingDirectory: '$(Build.SourcesDirectory)'
+
+  - task: VisualStudioTestPlatformInstaller@1
+    inputs:
+      versionSelector: latestStable
+
+  - task: PowerShell@2
+    displayName: Add test locale to User Language List
+    inputs:
+      targetType: 'inline'
+      script: |
+            $langToAdd = "${{ parameters.testLocale }}"
+            
+            Write-Host "Adding $langToAdd to user language list"
+            $langList = Get-WinUserLanguageList
+            $langList.Insert(0, $langToAdd)
+            Set-WinUserLanguageList -LanguageList $langList -Force
+
+            Write-Host "Get-WinUserLanguageList:" 
+            Get-WinUserLanguageList
+
+  - task: PowerShell@2
+    displayName: Display OS build/version/language info
+    inputs:
+      targetType: 'inline'
+      script: |
+            Get-Item -Path 'HKLM:\Software\Microsoft\Windows NT\CurrentVersion'
+            Write-Host (reg query "HKLM\SYSTEM\CurrentControlSet\Control\MUI" /s)
+            Write-Host (reg query "HKLM\SYSTEM\CurrentControlSet\Control\Nls" /s)
+            Write-Host (reg query "HKCU\Control Panel\International" /s)
+
+            Write-Host "Get-WmiObject -Class Win32_OperatingSystem" 
+            $OSInfo = Get-WmiObject -Class Win32_OperatingSystem
+            $languagePacks = $OSInfo.MUILanguages
+            $languagePacks
+
+            Write-Host "dism /online /Get-Intl" 
+            dism /online /Get-Intl
+
+            Write-Host "Get-WinUserLanguageList" 
+            Get-WinUserLanguageList
+
+            Write-Host "Get-WinUILanguageOverride"
+            Get-WinUILanguageOverride
+
+            Write-Host "Get-WinSystemLocale"
+            Get-WinSystemLocale
+  
+  - task: PowerShell@2
+    displayName: 'Run TAEF Tests'
+    inputs:
+      filePath: 'TestAll.ps1'
```

Comment:
This script today only detects/runs PushNotification's tests, right?
Are you aware of the status of any other existing tests, in terms whether they will pass/fail when being run in pipeline?

Reply:
I have tests I've been itching to light up but the test infrastructure had (significant) gaps needing love before I could (and I've been too busy to plug them myself). Paul's done an awesome job here not only plugging those gaps but simplifying the process to be practical for developers to use productively.

Lighting up my tests will still be work but with all this will be nearly trivial for me to do. _That_ I can find time for sooner than later :-)

In theory _most_ existing UT/FT can be lit up in the pipeline. A few have errors last I looked and/or may have errors when run in the pipeline and/or in all the test environments. More significantly, not all tests _should_ be run in all test passes -- e.g. we may not want all UT/FT to light up in PR/CI test phases. That's soluble with a little work (see other comment). Nothing that should hold up this PR, just lots of exciting opportunities for future PRs :D

Only real Q is if we want each team to make those calls or a more centralized effort by some/1 dev. I recommend the former as much as possible. The judgement if not-100% should run in all phases requires subject matter experts to make the right call. If the effort to make a .testdef is low (as it looks like) pinging teams to make a small investment to automate using existing tests should be an easy sell.

Reply:
This script detects any test folder with a testdef and looks for the component filename.dll described from the testdef. But I'm not sure of the status of all component testing whether they are stable tests or not when ran in the pipeline.

Reply:
Yeah, this is one of the items I'd like to follow up once this PR is in. Agreed that this is nonblocking.